test/ShallowWrapper-spec.js

```diff
@@ -158,6 +158,19 @@ describe('shallow', () => {
       expect(wrapper.contains(passes2)).to.equal(true);
     });
 
+    it('shouldn\'t confuse non React elements and string literals in DOM', () => {
+      const functionalComponent = () => <div></div>;
+
+      const wrapper = shallow(
+        <div>
+          <functionalComponent />
+          { 'Text node' }
+        </div>
+      );
+
+      expect(wrapper.contains(functionalComponent)).to.equal(false);
```

Comment:
I'm not sure how that changes this shallow rendering case tho - `functionalComponent` here is an SFC. Whether it's upper or lower case, why would anyone expect the wrapper not to contain something they just rendered in it?